File: code/617.go
/*
 * @Author: lmio
 * @Date: 2023-02-07 23:52:37
 * @LastEditTime: 2023-03-28 15:56:32
 * @FilePath: /leetcode/code/617.go
 * @Description:617.合并二叉树
 */
package code

import ."leetcode/code_struct" 

func MergeTrees(root1 *TreeNode, root2 *TreeNode) *TreeNode {
	if root1 == nil && root2 == nil {
		return nil
	}
	if root1 == nil {
		return root2
	}
	if root2 == nil {
		return root1
	}
	var node TreeNode
	node.Val = root1.Val + root2.Val
	node.Left = MergeTrees(root1.Left, root2.Left)
	node.Right = MergeTrees(root1.Right, root2.Right)
	return &node
}

File: code/617.py
'''
Author: lmio 2091319361@qq.com
Date: 2023-08-14 16:53:27
LastEditors: lmio 2091319361@qq.com
Description: 617. 合并二叉树
'''

from typing import Optional
from utils.node import TreeNode


class Solution:
    def mergeTrees(self, root1: Optional[TreeNode], root2: Optional[TreeNode]) -> Optional[TreeNode]:
        if not root1 and not root2:
            return None
        elif root1 and root2:
            node = TreeNode(root1.val + root2.val)
            node.left = self.mergeTrees(root1.left, root2.left)
            node.right = self.mergeTrees(root1.right, root2.right)
            return node
        elif root1:
            return root1
        else:
            return root2